functests/cluster_upgrade_test.go

```diff
@@ -0,0 +1,76 @@
+package functests_test
+
+import (
+	"flag"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	tests "github.com/openshift/ocs-operator/functests"
+
+	deploymanager "github.com/openshift/ocs-operator/pkg/deploy-manager"
+)
+
+var _ = Describe("Cluster upgrade", func() {
+	flag.Parse()
+
+	BeforeEach(func() {
+		RegisterFailHandler(Fail)
+	})
+
+	Describe("ocs", func() {
+
+		BeforeEach(func() {
+			By("Preparing for upgrade. Uninstall the current cluster")
+			tests.AfterTestSuiteCleanup()
+			By("Reinstall a fresh cluster based on upgrade_from image")
+			tests.BeforeUpgradeTestSuiteSetup()
+		})
+
+		Context("upgrade cluster", func() {
+			It("and verify deployment status", func() {
+				if tests.UpgradeFromOcsRegistryImage == "" || tests.UpgradeFromLocalStorageRegistryImage == "" {
+					Skip("Condition not met for upgrade")
+				}
+				deployManager, err := deploymanager.NewDeployManager()
+				Expect(err).To(BeNil())
+
+				By("Getting the ocs-catalogsource pod image before the upgrade")
+				ocsImage, err := deployManager.GetDeploymentImage("ocs-operator")
+
+				By("Getting the current csv before the upgrade")
+				csv, err := deployManager.GetCsv()
+				Expect(err).To(BeNil())
+				
+				By("Upgrading OCS with OLM to the current version from upgrade_from version")
+				err = deployManager.UpgradeOCSWithOLM(tests.OcsRegistryImage, tests.LocalStorageRegistryImage, tests.OcsSubscriptionChannel)
+				Expect(err).To(BeNil())
+
+				By("Waiting for OCS CSV to be posted and installed")
+				err = deployManager.WaitForCsvUpgrade(csv.Name, tests.OcsSubscriptionChannel)
+				Expect(err).To(BeNil())
+
+				By("Waiting for ocs-operator, rook-ceph-operator and noobaa-operator to come online.")
+				err = deployManager.WaitForOCSOperator()
```

Comment:
maybe something to think about here.

There might be a timing issue here where the deployments look like they're online simply because the new CSV hasn't been reconciled yet. This would make the next check were we cross reference the deployment images fail potentially.

Reply:
We wait until CSV reached "Succeeded" phase in the previous step, which means the resources were created successfully. "Installing" means install strategy has been initiated but not completed. I think that will cover it, but I see what you mean.

Reply:
ah, got it. learned something, thanks :+1: